### PR TITLE
SPARKC-480 Support load(path) for datasource

### PR DESCRIPTION
**Pull request by alexliu68**

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
My big question on this would be what do we gain from having another way of choosing the Cassandra Table? We already have one way so why add a second parallel method?

**Comment by alexliu68** [comment]
It's from one of DSE ticket to make it more easy to load the table similar way as load a file.

**Comment by RussellSpitzer** [comment]
My concern is that we would now have 3 ways to load a dataframe

```scala
spark.read.format("org.apache.spark.sql.cassandra").options(Map("table"-> "tab", "keyspace" -> "ks")
spark.read.cassandraFormat("tab", "ks")
spark.read.format("org.apache.spark.sql.cassandra").load("tab.ks")

```

**Comment by alexliu68** [comment]
load method is a public API from Spark, the pull is an implementation of the public API. As OSS Spark, there are multiple ways to load data sources. 

`spark.read.cassandraFormat("tab", "ks")` is more a custom api from SCC.
It's ok to have all different ways to load a table.

File: spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/sql/CassandraDataFrameSpec.scala [hunk 1]
```diff
@@ -81,13 +81,7 @@ class CassandraDataFrameSpec extends SparkCassandraITFlatSpecBase with Eventuall
...
       .read
       .format("org.apache.spark.sql.cassandra")
-      .options(
-        Map(
-          "table" -> "kv",
-          "keyspace" -> ks
-        )
-      )
-      .load()
+      .load(s"$ks.kv")
```
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/sql/CassandraDataFrameSpec.scala]: I think we should have a test where we override the "options" , maybe there should be a message if you do that?

Comment (alexliu68) [inline spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/sql/CassandraDataFrameSpec.scala]: I added one for it.

**Pull request closed without merging**